astroid/tests/unittest_inference.py

```diff
@@ -2303,6 +2303,27 @@ class A(object): pass
             error = errors[0]
             self.assertEqual(str(error), expected_value)
 
+    def test_unary_type_errors_crash(self):
+        ast_nodes = extract_node('''
```

Comment:
@degustaf what exactly is the problem here? Running pylint over this snipper raises indeed a TypeError, due to ``class_context`` being passed, but why was it passed to ``getattr`` in the first place? This pull request seems to address the effect rather than the cause. Let's investigate first in what circumstances ``class_context`` gets passed and why astroid does this.